yearn/contracts/SaddleStrategy.sol

```diff
@@ -0,0 +1,395 @@
+        // seems to confirm that.
+        return balanceOfWant();
+    }
+
+    /// @notice This method is defined in the BaseStrategy contract and is meant
+    ///         to do anything necessary to prepare this strategy for migration,
+    ///         such as transferring any reserve or LP tokens, CDPs, or other
+    ///         tokens or stores of value. This strategy implements the
+    ///         aforementioned behavior by withdrawing all vault's underlying
+    ///         tokens from the tBTC v2 Saddle reward pool.
+    /// @param newStrategy Address of the new strategy meant to replace the
+    ///        current one.
+    function prepareMigration(address newStrategy) internal override {
+        // Withdraw all the LP tokens from the tBTC v2 Saddle reward pool.
+        // These tokens will be transferred to the new strategy in the `migrate`
+        // function.
+        ILPRewardsTBTCV2Saddle(lpRewardsTbtcV2Saddle).withdraw(balanceOfPool());
+        // Get all the earned KEEP tokens and transfer them to the new strategy.
+        ILPRewardsTBTCV2Saddle(lpRewardsTbtcV2Saddle).getReward();
+        IERC20(keep).safeTransfer(
+            newStrategy,
+            IERC20(keep).balanceOf(address(this))
+        );
+        //TODO: Check if both the LP Token and KEEP token should be migrated to
+        // the new strategy, or just the LP token.
+    }
+
+    /// @notice This method is defined in the BaseStrategy contract and is meant
+    ///         to perform any strategy unwinding or other calls necessary to
+    ///         capture the free return this strategy has generated since the
+    ///         last time its core position(s) were adjusted. Examples include
+    ///         unwrapping extra rewards. This call is only used during normal
+    ///         operation of a strategy, and should be optimized to minimize
+    ///         losses as much as possible.
+    /// @param debtOutstanding Will be 0 if the strategy is not past the
+    ///        configured debt limit, otherwise its value will be how far past
+    ///        the debt limit the strategy is. The strategy's debt limit is
+    ///        configured in the vault.
+    /// @return profit Amount of realized profit.
+    /// @return loss Amount of realized loss.
+    /// @return debtPayment Amount of repaid debt. The value of debtPayment
+    ///         should be less than or equal to debtOutstanding. It is okay for
+    ///         it to be less than debtOutstanding, as that should only used as
+    ///         a guide for how much is left to pay back. Payments should be
+    ///         made to minimize loss from slippage, debt, withdrawal fees, etc.
+    function prepareReturn(uint256 debtOutstanding)
+        internal
+        override
+        returns (
+            uint256 profit,
+            uint256 loss,
+            uint256 debtPayment
+        )
+    {
+        /* solhint-disable not-rely-on-time */
```

Comment:
I think we don't need this. `not-rely-on-time` is considered as warning in `.solhint.json`.

Reply:
Removed in https://github.com/keep-network/tbtc-v2/pull/13/commits/5267c11fefbf0a07b3e4b19a6af70b943e735bf6.